python/tests/unit/utils/data_splitters_test.py

```diff
@@ -504,8 +510,50 @@ def test_node_based_link_splitter_parallelized(self):
             args=(
                 3,  # world_size
                 init_method,  # init_method
-                edges,  # edges
+                edges,  # tensors
+                expected_splits,  # expected
+                False,  # is_node_split
+            ),
+            nprocs=3,
+            join=True,
+        )
+
+    def test_node_based_splitter_parallelized(self):
+        init_method = get_process_group_init_method()
+        nodes = [
+            torch.arange(0, 20, dtype=torch.int64),
+            torch.arange(0, 10, dtype=torch.int64),
+            torch.arange(0, 20, 2, dtype=torch.int64),
+        ]
+        # We need to guarantee that a given node id is always selected into the same split, on every rank.
+        expected_splits = [
+            (
+                torch.arange(16, dtype=torch.int64),
+                torch.tensor([16, 17], dtype=torch.int64),
+                torch.tensor([18, 19], dtype=torch.int64),
```

Comment:
comments would help reader.
```suggestion
                torch.arange(16, dtype=torch.int64),  # train_split on process_1: using identity hash, we expect 80% of data  .... etc.
                torch.tensor([16, 17], dtype=torch.int64), # test_split on process_1; 
                torch.tensor([18, 19], dtype=torch.int64),
```